src/components/devices/DeviceList.vue

```diff
@@ -1,55 +1,88 @@
-        <div class="list-group-item" v-for="item in items" :key="item.id">
-          <div class="row">
-            <div class="col align-self-center">
-              <router-link to="/devices/1">{{ item.name }}</router-link>
-            </div>
-            <div class="col align-self-center">
-              <div class="text-muted">{{ item.description }}</div>
-            </div>
-            <div class="col text-right">
-              <router-link
-                class="btn btn-sm btn-primary"
-                :to="{ name: 'DeviceUpdate', params: { id: item.id } }"
-              >
-                <i class="fas fa-pencil-alt mr-2"></i>
-                Edit
+  <div class="card mb-4">
+    <TableHead :name="title" :links="tableHeadLinks" />
+    <TableBodyNoData v-if="items.length === 0" />
+    <div v-else class="list-group list-group-flush">
+      <div class="list-group-item" v-for="(item, index) in items" :key="index">
+        <div class="row">
+          <div class="col align-self-center">
+            <router-link :to="{ name: 'DeviceDetails', params: { id: item.id } }">{{
+              item.label
+            }}</router-link>
+          </div>
+          <div class="col align-self-center">
+            <div v-if="!item.event" class="text-muted">N/A</div>
+            <div v-else>
+              Connected to
+              <router-link :to="{ name: 'EventOverview', params: { id: item.event.id } }">
+                {{ item.event.name }}
               </router-link>
             </div>
           </div>
+          <div class="col text-right">
+            <router-link
+              class="btn btn-sm btn-primary"
+              :to="{ name: 'DeviceUpdate', params: { id: item.id } }"
+            >
+              <i class="fas fa-pencil-alt mr-2"></i>
+              Edit
+            </router-link>
+          </div>
         </div>
       </div>
     </div>
   </div>
 </template>
 
 <script>
-// import store from '@/store/data';
+import { mapState, mapActions } from 'vuex';
+import TableHead from '@/components/_ui/TableHead.vue';
+import TableBodyNoData from '@/components/_ui/TableBodyNoData.vue';
 
 export default {
+  name: 'DeviceList',
+  components: { TableHead, TableBodyNoData },
   data() {
     return {
-      items: [],
+      title: 'Devices',
+      tableHeadLinks: [
+        {
+          name: 'Add',
+          routeName: 'DeviceCreate',
+          class: 'btn-primary',
+          icon: 'fa-plus',
+        },
+      ],
     };
   },
   created() {
-    // this.items = store.devices;
+    this.setCurrentHeader('...');
   },
   mounted() {
-    this.$store.dispatch('application/setHeaderInfo', {
-      title: 'Devices',
-      subtitle: 'Total: 0',
-    });
+    (async () => {
+      try {
+        await this.fetchItems();
+        this.setCurrentHeader(`Total: ${this.itemsTotal}`);
+      } catch (error) {
+        console.log('[Exception-DeviceList]', error);
+      }
+    })();
```

Comment:
Please make mounted async instead of creating an IIFE inside the method.

Reply:
Theoretically 'mounted' hook was designed to be synchronous not asynchronous, but I can change this in the next commit.